### PR TITLE
Upgrade to webpack 4 and webpack-dev-server 3

### PR DESCRIPTION
**Pull request by rmars**
Ugraded to webpack 4 in the hopes of speeding up build times. Now the build is slower. Comments welcome on improving this config.

```
Time: 27002ms
```

### CODE REVIEW

**Comment by rmars** [comment]
Adding `mode: process.env.NODE_ENV` has speeded this up to an acceptable 
```
Time: 1019ms
```
Thanks @grampelberg!

File: web/app/webpack.config.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 /* global require, module, __dirname */
 
 const path = require('path');
+const webpack = require('webpack');
```
Comment (grampelberg) [inline web/app/webpack.config.js]: Not used.

File: web/app/webpack.config.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+var HtmlWebpackPlugin = require('html-webpack-plugin');
```
Comment (grampelberg) [inline web/app/webpack.config.js]: `const` instead?

File: web/app/package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "d3": "^4.11.0",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
+    "html-webpack-plugin": "^3.2.0",
```
Comment (grampelberg) [inline web/app/package.json]: This is a devDependency.

**Comment by rmars** [comment]
Hmmmm I added the HtmlWebpackPlugin when I was messing around with trying to make the build faster, but I don't think it is necessary

File: web/app/webpack.config.js (hunk 1 above)
Comment (grampelberg) [reply web/app/webpack.config.js]: This should go away right?

**Pull request merged**